keps/2937-resource-transformer/README.md

```diff
@@ -0,0 +1,405 @@
+avoids conflicting definitions of `Replace` for an input resource.
+
+### Option 2: Singleton Matrix Output Focused
+This is similar to Option 1, but makes the output resource the
+primary key.  
+```go
+type ResourceTransformationStrategy string
+const Retain ResourceTransformationStrategy = "Retain"
+const Replace ResourceTransformationStrategy = "Replace"
+
+type ResourceTransformationMatrix struct {
+  // The transformations to apply where the first map key is the output resource name
+  // and the inner map key is the input resource name
+  Transformations map[corev1.ResourceName]map[corev1.ResourceName]resource.Quantity `json:"transformations"`
+
+  // The map key is the input resource name.
+  Strategy map[corev1.ResourceName]ResourceTransformationStrategy `json:"strategy"`
+}
+```
+
+Grouping by output makes it easier for the cluster admin to see/specify
+all the resources that map into a given output as a unit.  However it
+comes at the cost of a separate map to specify strategy.
+It may be confusing that the `Transformations` map is keyed by
+output name as the primary key while `Strategy` is keyed by input name.
+
+### Comparison
+
+Consider the following list of 4-tuples
+
+* (`nvidia.com/mig-1g.5gb`, `kueue.x-k8s.io/accelerator-memory`, `5G`, `replace`)
+* (`nvidia.com/mig-2g.10gb`, `kueue.x-k8s.io/accelerator-memory`, `10G`, `replace`)
+* (`nvidia.com/mig-4g.20gb`, `kueue.x-k8s.io/accelerator-memory`, `20G`, `replace`)
+* (`nvidia.com/mig-1g.5gb`, `kueue.x-k8s.io/credits`, `10`, `replace`)
+* (`nvidia.com/mig-2g.10gb`, `kueue.x-k8s.io/credits`, `15`, `replace`)
+* (`nvidia.com/mig-4g.20gb`, `kueue.x-k8s.io/credits`, `30`, `replace`)
+* (`cpu`, `kueue.x-k8s.io/credits`, `1`, `retain`)
+
+Option 1
+```yaml
+resourceTransformationMatrix:
+  transformations:
+    nvidia.com/mig-1g.5gb:
+      strategy: Replace
+      conversionBaseValues:
+        kueue.x-k8s.io/accelerator-memory: 5G
+        kueue.x-k8s.io/credits: 10
+    nvidia.com/mig-2g.10gb:
+      strategy: Replace
+      conversionBaseValues:
+        kueue.x-k8s.io/accelerator-memory: 10G
+        kueue.x-k8s.io/credits: 15
+    nvidia.com/mig-4g.20gb:
+      strategy: Replace
+      conversionBaseValues:
+        kueue.x-k8s.io/accelerator-memory: 20G
+        kueue.x-k8s.io/credits: 30
+    cpu:
+      strategy: Retain
+      conversionBaseValues:
+        kueue.x-k8s.io/credits: 1
+```
+
+Option 2
+```yaml
+resourceTransformationMatrix:
+  transformations:
+    kueue.x-k8s.io/accelerator-memory:
+      nvidia.com/mig-1g.5gb: 5G
+      nvidia.com/mig-2g.10gb: 10G
+      nvidia.com/mig-4g.20gb: 20G
+    kueue.x-k8s.io/credits:
+      nvidia.com/mig-1g.5gb: 10
+      nvidia.com/mig-2g.10gb: 15
+      nvidia.com/mig-4g.20gb: 30
+      cpu: 1
+  strategy:
+    nvidia.com/mig-1g.5gb: Replace
+    nvidia.com/mig-2g.10gb: Replace
+    nvidia.com/mig-4g.20gb: Replace
+    cpu: Retain
+```
+
+Option 2 is likely to be terser, but I find Option 1 slightly more intutive because
```

Comment:
+1 to option 1

Reply:
Since everyone who commented liked option 1 for specifying the configuration better than option 2, I'm going to simplify the KEP by just proposing option 1.